server/src/test/java/org/elasticsearch/indices/IndicesRequestCacheIT.java

```diff
@@ -185,29 +188,27 @@ public void testQueryRewriteMissingValues() throws Exception {
                 .setSettings(Settings.builder().put(IndicesRequestCache.INDEX_CACHE_REQUEST_ENABLED_SETTING.getKey(), true)
-                        .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1).put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)).get());
-        indexRandom(true, client.prepareIndex("index", "type", "1").setSource("d", "2014-01-01T00:00:00"),
-                client.prepareIndex("index", "type", "2").setSource("d", "2014-02-01T00:00:00"),
-                client.prepareIndex("index", "type", "3").setSource("d", "2014-03-01T00:00:00"),
-                client.prepareIndex("index", "type", "4").setSource("d", "2014-04-01T00:00:00"),
-                client.prepareIndex("index", "type", "5").setSource("d", "2014-05-01T00:00:00"),
-                client.prepareIndex("index", "type", "6").setSource("d", "2014-06-01T00:00:00"),
-                client.prepareIndex("index", "type", "7").setSource("d", "2014-07-01T00:00:00"),
-                client.prepareIndex("index", "type", "8").setSource("d", "2014-08-01T00:00:00"),
-                client.prepareIndex("index", "type", "9").setSource("d", "2014-09-01T00:00:00"));
-        ensureSearchable("index");
-        assertCacheState(client, "index", 0, 0);
+                    .put(INDEX_REFRESH_INTERVAL_SETTING.getKey(), "-1") // disable automatic refreshes
+                    .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1).put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)).get());
 
-        // Force merge the index to ensure there can be no background merges during the subsequent searches that would invalidate the cache
-        ForceMergeResponse forceMergeResponse = client.admin().indices().prepareForceMerge("index").setFlush(true).get();
-        ElasticsearchAssertions.assertAllSuccessful(forceMergeResponse);
-        refresh();
+        BulkRequestBuilder bulkBuilder = client().prepareBulk();
+        bulkBuilder.add(client.prepareIndex("index", "type", "1").setSource("d", "2014-01-01T00:00:00"));
+        bulkBuilder.add(client.prepareIndex("index", "type", "2").setSource("d", "2014-02-01T00:00:00"));
+        bulkBuilder.add(client.prepareIndex("index", "type", "3").setSource("d", "2014-03-01T00:00:00"));
+        bulkBuilder.add(client.prepareIndex("index", "type", "4").setSource("d", "2014-04-01T00:00:00"));
+        bulkBuilder.add(client.prepareIndex("index", "type", "5").setSource("d", "2014-05-01T00:00:00"));
+        bulkBuilder.add(client.prepareIndex("index", "type", "6").setSource("d", "2014-06-01T00:00:00"));
+        bulkBuilder.add(client.prepareIndex("index", "type", "7").setSource("d", "2014-07-01T00:00:00"));
+        bulkBuilder.add(client.prepareIndex("index", "type", "8").setSource("d", "2014-08-01T00:00:00"));
+        bulkBuilder.add(client.prepareIndex("index", "type", "9").setSource("d", "2014-09-01T00:00:00"));
+        BulkResponse actionGet = bulkBuilder.execute().actionGet();
```

Comment:
nit: `actionGet` isn't a good name ;)

Reply:
Thanks @dnhatn, addressed in 7f125e75efd8f697a6fae8820e9712ed8b36a25d